katharsis-core/src/main/java/io/katharsis/resource/registry/repository/adapter/RelationshipRepositoryAdapter.java

```diff
@@ -0,0 +1,290 @@
+
+			@Override
+			protected JsonApiResponse invoke(RepositoryFilterContext context) {
+				RepositoryRequestSpec request = context.getRequest();
+				Object source = request.getEntity();
+				Iterable<?> targetIds = request.getIds();
+				String fieldName = request.getRelationshipField();
+				QueryAdapter queryAdapter = request.getQueryAdapter();
+		        if (isAnnotated) {
+		            ((AnnotatedRelationshipRepositoryAdapter) relationshipRepository)
+		                .addRelations(source, targetIds, fieldName, queryAdapter);
+		        } else if(relationshipRepository instanceof QuerySpecRelationshipRepository){
+		        	((QuerySpecRelationshipRepository)relationshipRepository).addRelations(source, targetIds, fieldName);
+		        } else {
+		            ((RelationshipRepository) relationshipRepository).addRelations(source, targetIds, fieldName);
+		        }
+		        return new JsonApiResponse();
+			}
+		};
+		RepositoryRequestSpec requestSpec = RepositoryRequestSpecImpl.forRelation(moduleRegistry.getResourceRegistry(), HttpMethod.POST, source, queryAdapter, targetIds, fieldName, resourceInformation.getResourceClass());
+		return chain.doFilter(newRepositoryFilterContext(requestSpec));
+    }
+
+    @SuppressWarnings("rawtypes")
+	public JsonApiResponse removeRelations(T source, Iterable<J> targetIds, String fieldName, QueryAdapter queryAdapter) {
+    	RepositoryRequestFilterChainImpl chain = new RepositoryRequestFilterChainImpl() {
+
+			@Override
+			protected JsonApiResponse invoke(RepositoryFilterContext context) {
+				RepositoryRequestSpec request = context.getRequest();
+				Object source = request.getEntity();
+				Iterable<?> targetIds = request.getIds();
+				String fieldName = request.getRelationshipField();
+				QueryAdapter queryAdapter = request.getQueryAdapter();
+		    	if (isAnnotated) {
+		            ((AnnotatedRelationshipRepositoryAdapter) relationshipRepository)
+		                .removeRelations(source, targetIds, fieldName, queryAdapter);
+		        } else if(relationshipRepository instanceof QuerySpecRelationshipRepository){
+		        	((QuerySpecRelationshipRepository)relationshipRepository).removeRelations(source, targetIds, fieldName);
+		        } else {
+		            ((RelationshipRepository) relationshipRepository).removeRelations(source, targetIds, fieldName);
+		        }
+		        return new JsonApiResponse();
+			}
+		};
+		RepositoryRequestSpec requestSpec = RepositoryRequestSpecImpl.forRelation(moduleRegistry.getResourceRegistry(), HttpMethod.DELETE, source, queryAdapter, targetIds, fieldName, resourceInformation.getResourceClass());
+		return chain.doFilter(newRepositoryFilterContext(requestSpec));
+    }
+
+	@SuppressWarnings("rawtypes")
+	public JsonApiResponse findOneTarget(I sourceId, String fieldName, QueryAdapter queryAdapter) {
+    	RepositoryRequestFilterChainImpl chain = new RepositoryRequestFilterChainImpl() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 24 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1188)